tests/e2e/utils/flows.js

```diff
@@ -832,4 +832,66 @@ export const merchantWCP = {
 		}
 		return wasInitiallyEnabled;
 	},
+
+	addMulticurrencyWidget: async () => {
+		await merchantWCP.openWCPSettings();
+		await merchantWCP.activateMulticurrency();
+		await page.goto( `${ config.get( 'url' ) }wp-admin/widgets.php`, {
+			waitUntil: 'networkidle0',
+		} );
+		await uiLoaded();
+
+		const closeWelcomeModal = await page.$( 'button[aria-label="Close"]' );
+		if ( closeWelcomeModal ) {
+			await closeWelcomeModal.click();
+		}
+
+		const isWidgetAdded = await page.$(
+			'.wp-block iframe[srcdoc*=\'name="currency"\']'
+		);
+		if ( ! isWidgetAdded ) {
+			await page.click( 'button[aria-label="Add block"]' );
+
+			const searchInput = await page.waitForSelector(
+				'input.components-search-control__input'
+			);
+			searchInput.type( 'switcher', { delay: 20 } );
+
+			await page.waitForSelector(
+				'button.components-button[role="option"]'
+			);
+			await page.click( 'button.components-button[role="option"]' );
+			await page.waitFor( 2000 );
+			await page.waitForSelector(
+				'.edit-widgets-header .edit-widgets-header__actions button.is-primary'
+			);
+			await page.click(
+				'.edit-widgets-header .edit-widgets-header__actions button.is-primary'
+			);
+			await expect( page ).toMatchElement( '.components-snackbar', {
+				text: 'Widgets saved.',
+				timeout: 15000,
+			} );
+			await page.waitFor( 1000 );
```

Comment:
A bit of both. It was there because during the development it was needed for the test to pass. After removing it today I could see the tests are stable without it. Done in https://github.com/Automattic/woocommerce-payments/pull/7822/commits/230356e66ec66c304560795bc59ef644d974ff29.